storage/src/backend/registry.rs

```diff
@@ -487,8 +517,30 @@ impl RegistryReader {
                 return self._try_read(buf, offset, false);
             }
         } else {
-            resp =
-                self.request::<&[u8]>(Method::GET, url.as_str(), None, headers.clone(), false)?;
+            resp = match self.request::<&[u8]>(
+                Method::GET,
+                url.as_str(),
+                None,
+                headers.clone(),
+                false,
+            ) {
+                Ok(res) => res,
+                Err(RegistryError::Request(ConnectionError::Common(e)))
+                    if &error_source2_string(&e)[0..14] == "error:1408F10B" =>
+                {
+                    if self.state.scheme.0.load(Ordering::Relaxed) {
+                        self.state.scheme.0.store(false, Ordering::Relaxed);
```

Comment:
How about putting this logic into a helper func like `self.state.fallback_http`.